### DISCUSSION TITLE
Pimcore REST API for create an Dataobject

### DISCUSSION THREAD

**Discussion by bhagya7977** [pimcore/pimcore#17171] [Q&A / Support]
I am creating an REST API to create an dataobject  and   given the Body as JSON  with below properties and  My dataobject is contains the fields  name , description , price  
{
    "name": "Test product",
    "description": "This product is creating via the POST API ",
    "price": 50
}

while hitting the POST API , throwing an below error like 
![image](https://github.com/pimcore/pimcore/assets/105700672/80013be6-1f03-4a70-a13e-06ac407b2ecc)

**Comment by kingjia90** [comment]
the `key` is missing, and maybe also the `parent`
```
{
"name": "Test product",
"description": "This product is creating via the POST API ",
"price": 50
"key" "something"
}
```
see https://pimcore.com/docs/platform/2023.1/Pimcore/Objects/Working_with_PHP_API/#crud-operations

**Reply by bhagya7977** [reply]
yes i have passed the  Key and parentID  in the Json  able to create an New Dataobject .

**Reply by bhagya7977** [reply]
I have one more query  , I have added the  sku field as input parameter to the Product Dataobject . I am able to fetch the product object using the  id  param , But if I have trying to load the  Product object with the param  SKU , its not returning the  product object. Can you please suggest me the approach to fetch data by sku param ? 

Below is the code to fetch the data  by SKu  with below ways  

     `$products` = new DataObject\DemoProduct\Listing();
        $products->setCondition("sku= ?", $data['sku']);  
        
        or  
        $product = DataObject\DemoProduct::getBySku($data['sku']);

**Reply by kingjia90** [reply]
iirc there would be no `getBySku()` 🤔 so the listing approach would be fine and should be working but can't tell from the snippet

**Reply by bhagya7977** [reply]
Is there a way to get the output  with field's name, SKU, description  only in the product data object listing output .  While we are printing the Product Dataobject coming with the all protected properties and inherited methods.

**Reply by kingjia90** [reply]
Mhhh, do you mean like a sort of `SELECT` of sql? not that i am aware of.

You get the full object from the listing and you just need to loop through this 
```
foreach ($products as $product){ 
    $name = $product->getName();
    $sku = $product->getSku();
    $desc = $product->getDescription();
}
```

**Reply by bhagya7977** [reply]
I have tried with the above piece of code to get the Product object with specified fields, I want to filter the collection with specific parameter like SKU to get the collection output .

**Reply by bhagya7977** [reply]
I have prepared an data  by using the param   SKU  and also able to update the Dataobject . 
$products = DataObject\DemoProduct::getBySku($data['sku']); 

        foreach ($products as $key => $product) {
            $data[] = array(
                "id" => $product->getId()  ,
                "price" => $product->getPrice()  ,
            );
        }

**Comment by bhagya7977** [comment]
issue resolved and Able to create & Update Dataobjects pro-grammatically via REST API
